src/main/java/ro/hiringsystem/service/CandidateUserService.java

```diff
@@ -0,0 +1,25 @@
+package ro.hiringsystem.service;
+
+import ro.hiringsystem.model.CandidateUser;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+public interface CandidateUserService {
```

Comment:
The code is very repetitive, mostly varying by the type of User. All the service interfaces for them could be replaced with a template interface.
```
public interface UserService<T extends User> {
      Optional<T> getById(UUID id);
      ...
      etc
}

public interface CandidateUserService extends UserService<CandidateUser>{
        //some specific method for CandidateUserService
}
```